src/olympia/reviewers/views.py

```diff
@@ -1511,20 +1511,21 @@ def get_objects(self):
 
     def retrieve(self, request, *args, **kwargs):
         objs = self.get_objects()
+        version = objs['instance']
 
         if self.request.GET.get('file_only', False):
             serializer = FileEntriesDiffSerializer(
-                instance=objs['instance'].current_file,
+                instance=version.current_file,
                 context={
                     'exclude_entries': True,
                     'file': self.request.GET.get('file', None),
                     'request': self.request,
                     'parent_version': objs['parent_version'],
                 })
-            return Response({'file': serializer.data})
+            return Response({'id': version.id, 'file': serializer.data})
 
         serializer = AddonCompareVersionSerializer(
-            instance=objs['instance'],
+            instance=version,
```

Comment:
same comments on this class.  I know it's more work but these small quick hacks are turning into something more complicated.

Reply:
actually, I think this one _does_ need a custom retrieve model because it uses `get_objects` rather than `get_object`.  For maintainability it would be best to still have a `get_serializer` and `get_serializer_context` thought.